documentation/src/components/side-nav-search.ts

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+    LitElement,
+    html,
+    CSSResultArray,
+    TemplateResult,
+    property,
+} from 'lit-element';
+import sideNavSearchMenuStyles from './side-nav-search.css';
+import lunr from 'lunr';
+import { Search } from '../../../packages/search';
+import {
+    OverlayOpenDetail,
+    OverlayCloseDetail,
+} from '../../../packages/overlay-root';
+import { AppRouter } from '../router';
+
+const searchIndex = require('../../components/searchIndex.json');
+
+export function nameToTitle(name: string): string {
+    return name.replace(/((^|\-)(\w))/gm, (match, p1, p2, p3) => {
+        let result = p3.toUpperCase();
+        if (p2 === '-') {
+            result = ` ${result}`;
+        }
+        return result;
+    });
+}
+
+interface Result {
+    name: string;
+    url: string;
+}
+
+interface ResultGroup {
+    name: string;
+    results: Result[];
+    maxScore: number;
+}
+
+class SearchComponent extends LitElement {
+    private index: lunr.Index | undefined;
+
+    public static get styles(): CSSResultArray {
+        return [sideNavSearchMenuStyles];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public open: boolean = false;
+
+    @property({ type: Array })
+    public results: ResultGroup[] = [];
+
+    private popover?: HTMLElement | null;
+
+    private handleSearchInput(event: InputEvent) {
+        if (event.target) {
+            const searchField = event.target as Search;
+            this.updateSearchResults(searchField.value);
+        }
+    }
+
+    private openPopover() {
```

Comment:
We should nut out an API for doing programmatically triggered overlays (as opposed to using `overlay-trigger`).

Reply:
See #321